### ISSUE TITLE
Using minio(RELEASE.2022-06-20T23-13-45Z ) and mc(RELEASE.2022-06-26T18-51-48Z) version , Can't curl to api normally, after enable tls

### ISSUE THREAD

**Issue by wangglbj** [minio/minio#15765]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

Use the following minio/mc version curl api to report an error：

```
containers:
      - name: minio
        image: minio/minio:RELEASE.2022-06-20T23-13-45Z
        imagePullPolicy: IfNotPresent
        args:
        - server
        - /storage
        - --console-address
        - ":9001"
        env:
...
- name: mc
        image: minio/mc:RELEASE.2022-06-26T18-51-48Z
        imagePullPolicy: IfNotPresent
        command:
        - /bin/sh
        - -c
        - "mc --config-dir=/config config host add velero http://minio:9000 minio minio123 && mc --config-dir=/config mb -p 
velero/velero"
---
kind: Route
apiVersion: route.openshift.io/v1
metadata:
  name: minio-api
  namespace: velero
spec:
  to:
    kind: Service
    name: minio
    weight: 100
  port:
    targetPort: api
  tls:
    termination: edge
    insecureEdgeTerminationPolicy: None
  wildcardPolicy: None

# curl https://minio-api-velero.apps.symt01.cp.com:443/minio/health/cluster -k -v
 SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.apps.symt01.cp.com
*  start date: Aug 25 11:59:36 2022 GMT
*  expire date: Aug 24 11:59:37 2024 GMT
*  issuer: CN=ingress-operator@1661428777
*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
* TLSv1.3 (OUT), TLS app data, [no content] (0):
> GET /minio/health/cluster HTTP/1.1
> Host: minio-api-velero.apps.symt01.cp.com
> User-Agent: curl/7.61.1
> Accept: */*
> 
* TLSv1.3 (IN), TLS handshake, [no content] (0):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, [no content] (0):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS app data, [no content] (0):
< HTTP/1.1 503 Service Unavailable
< accept-ranges: bytes
< content-length: 0
< content-security-policy: block-all-mixed-content
< server: MinIO
< strict-transport-security: max-age=31536000; includeSubDomains
< vary: Origin
< x-amz-request-id: 171900F39FA80DAF
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< date: Wed, 28 Sep 2022 10:56:23 GMT
< set-cookie: b7f66f8f084b9012d13ad054b9d15c98=f4607f6e3e9eb25c911fce5680812f28; path=/; HttpOnly; Secure; SameSite=None
< 
* Connection #0 to host minio-api-velero.apps.symt01.cp.com left intact


```

## Expected Behavior
But it's normal to use the latest version with mino/mc by curl with tls

```
 containers:
      - name: minio
        image: minio/minio:latest
        imagePullPolicy: IfNotPresent
        args:
        - server
        - /storage
        - --console-address
        - ":9001"
- name: mc
        image: minio/mc:latest
        imagePullPolicy: IfNotPresent
        command:
        - /bin/sh
        - -c
        - "mc --config-dir=/config config host add velero http://minio:9000 minio minio123 && mc --config-dir=/config mb -p velero/velero"
       
curl https://minio-api-velero.apps.symt01.cp.com:443/minio/health/cluster -k -v

* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.apps.symt01.cp.com
*  start date: Aug 25 11:59:36 2022 GMT
*  expire date: Aug 24 11:59:37 2024 GMT
*  issuer: CN=ingress-operator@1661428777
*  SSL certificate verify result: self signed certificate in certificate chain (19), continuing anyway.
* TLSv1.3 (OUT), TLS app data, [no content] (0):
> GET /minio/health/cluster HTTP/1.1
> Host: minio-api-velero.apps.symt01.cp.com
> User-Agent: curl/7.61.1
> Accept: */*
> 
* TLSv1.3 (IN), TLS handshake, [no content] (0):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, [no content] (0):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS app data, [no content] (0):
< HTTP/1.1 200 OK
< accept-ranges: bytes
< content-length: 0
< content-security-policy: block-all-mixed-content
< server: MinIO
< strict-transport-security: max-age=31536000; includeSubDomains
< vary: Origin
< x-amz-request-id: 1719009210D8EE29
< x-content-type-options: nosniff
< x-minio-write-quorum: 1
< x-xss-protection: 1; mode=block
< date: Wed, 28 Sep 2022 10:49:24 GMT
< set-cookie: b7f66f8f084b9012d13ad054b9d15c98=db8be5c6269f103f7a15eceea52b0cc4; path=/; HttpOnly; Secure; SameSite=None
< cache-control: private
< 
* Connection #0 to host minio-api-velero.apps.symt01.cp.com left intact
```

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by wangglbj** [comment]
Is there a way to solve this problem? Since I don't  want to upgrade to the latest version currently. thanks

**Comment by klauspost** [comment]
AFAICT Already fixed in #15415
